### PR TITLE
WIP: Pre commit task

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies
https://trello.com/c/aba5f6D3/536-8-start-using-type-checkers-nokr-kr36

## This Pull Request implements

Add new task for running pre-commit.

## Description
I based this task on the coala-ci task.
Runs `pre-commit run --all-files` on working branch and outputs either success or failur.

### CODE REVIEW

File: tasks/pre-commit-check.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      workingDir: /workspace/repo
+      securityContext:
+        privileged: true
+      script: |
+        cp -r /workspace/pr/ /workspace/output/
+        git fetch origin pull/$(params.pr_number)/head:workbranch
+        git checkout workbranch
+
+    - name: run-pre-commit
+      image: quay.io/thoth-station/thoth-coala:v0.10.2  # TODO: replace with image for pre-commit
```
Comment (harshad16) [inline tasks/pre-commit-check.yaml:37]: Awesome work.
The build of the pre-commit can be done here: https://github.com/thoth-station/zuul-infra

**Comment by KPostOffice** [comment]
Depends on https://github.com/thoth-station/zuul-infra/pull/25

**Comment by KPostOffice** [comment]
@harshad16 Now that we have a pre-commit build WDYT?

File: tasks/pre-commit-check.yaml (hunk 1 above)
Comment (harshad16) [reply tasks/pre-commit-check.yaml:37]:
```suggestion
      image: quay.io/thoth-station/thoth-precommit:v0.10.3 
```

**Comment by harshad16** [comment]
> @harshad16 Now that we have a pre-commit build WDYT?

Point the image to [thoth-precommit](https://quay.io/repository/thoth-station/thoth-precommit), we can merge this. 
Thanks for the work :medal_military:

**Pull request merged**